Adapt our GC to v2.098

Geod24 opened the pull request:
The interface was changed in v2.098, so we get a 'does not implement function...' error.

Geod24 committed:
Adapt our GC to v2.098

The interface was changed in v2.098, so we get a 'does not implement function...' error.

hewison-chris commented on source/agora/utils/gc/GC.d:
```diff
@@ -408,8 +408,9 @@ class ConservativeGC : GC
-    BlkInfo qalloc( size_t size, uint bits, const TypeInfo ti) nothrow
+    static if (__VERSION__ <= 2097)
+    {
+    override BlkInfo qalloc( size_t size, uint bits, const TypeInfo ti) nothrow
```
Indentation missing?

Geod24 replied:
It was intentional, but I could also add it if you prefer.

hewison-chris replied:
Was that because it is `static if`? Other places we still indent.

hewison-chris reviewed (approved):
LGTM just think maybe you missed indentation in the static if statement. Or was it intentional?

Geod24 commented:
Looks like a bunch of other things are broken, great.

Geod24 commented:
https://github.com/vibe-d/vibe-core/pull/295

Tested locally, no more errors!

Geod24 committed:
Update vibe-core to a v2.098-compatible commit

We had to switch to our fork because of a regression in v2.098.0 (filed as 22300),
which we could not work around. Our fork is simply the current master branch with
all deprecated symbols removed.

Geod24 commented:
Rebased with upstream commit, fixed indent.

Geod24 commented:
> /opt/hostedtoolcache/dc/ldc2-36198b36/x64/ldc2-36198b36-linux-x86_64/bin/../import/core/internal/dassert.d(194,36): Error: template instance `opCast!(const(ManualEvent))` does not match template declaration `opCast(T : bool)()`

Okay I forgot `-checkaction=context` while testing `-.-`

Geod24 commented:
Reported as https://issues.dlang.org/show_bug.cgi?id=22300

Geod24 commented:
Third time's the charm, and that's why we use git submodules.
Don't forget to `git submodule sync; git submodule update` after pulling this.

kchulj reviewed (approved):
🆗

Geod24 commented:
Well, it compiles.
